release-20.2: sql: don't overwrite ctx with nil on failure

asubiotto opened the pull request:
Backport 1/1 commits from #59002.

/cc @cockroachdb/release

---

SetupLocalSyncFlow would return a nil ctx with an error in setupFlows if an
error occurred. The previous version would overwrite a ctx in the outer scope
just to return it. This commit no longer overwrites ctx and simply returns the
new ctx when there is no error. On master, this nil ctx would not have any
consequences although this commit is good hygiene, but 20.2 uses this ctx in a
defer, which would cause a panic.

Release note (bug fix): Fix a nil pointer panic edge case in query setup code.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59003)
<!-- Reviewable:end -->

Commit:
sql: don't overwrite ctx with nil on failure

SetupLocalSyncFlow would return a nil ctx with an error in setupFlows if an
error occurred. The previous version would overwrite a ctx in the outer scope
just to return it. This commit no longer overwrites ctx and simply returns the
new ctx. On master, this nil ctx would not have any consequences although this
commit is good hygiene, but 20.2 uses this ctx in a defer, which would cause a
panic.

Release note (bug fix): Fix a nil pointer panic edge case in query setup code.

joshimhoff commented:
This is still causing panics on that cloud cluster. I know we're all super busy these days and I know it's Friday but a merge today would be good. Leaving a silence in place for a while as we must do until the fix lands on the cluster is not ideal.

asubiotto commented:
Yep, just waiting on an LGTM on #59002 and this should go in pretty soon after (definitely today).

joshimhoff commented:
Thx much appreciated.

yuzefovich reviewed (approved):
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59003)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
